### ISSUE TITLE
UDAs breaking on empty input tables

### ISSUE THREAD

**Issue by okennedy** [xerial/sqlite-jdbc#276]
It took a little while to track this down, but as far as I can tell, Java-land User-Defined Aggregates are causing a SQLite assertion to break.  We define UDAs GROUP_AND and GROUP_OR ( the aggregate counterparts of `&&` and `||`).  

We can run GROUP_AND on a non-empty relation (`DIRECT` is a directive to pass-through to SQLite)):
```
mimir> direct (select group_and(0) as x from (select 1 from foo where 1=1) test; );
 x
-----
 '0'
```

We can run COUNT(*) on an empty relation
```
mimir> direct (select count(*) as x from (select 1 from foo where 1=0) test; );
 x
-----
 '0'
```

But when we try to run GROUP_AND on an empty relation we get:
```
mimir> direct (select group_and(0) as x from (select 1 from foo where 1=0) test; );
Assertion failed: (*func), function xFinal, file src/main/java/org/sqlite/core/NativeDB.c, line 378.
./bin/mimir-debug: line 8: 59885 Abort trap: 6           $CMD $*
```

For reference, GROUP_AND is defined here: https://github.com/UBOdin/mimir/blob/ReadingsDemo/src/main/scala/mimir/sql/sqlite/SQLiteCompat.scala#L377

**Comment by tildedave** [comment]
I've just run into this as well.  It looks like if xStep will initiate this, but if xFinal is executed without xStep (empty table), it crashes. 

What looks like correct allocation here: https://github.com/xerial/sqlite-jdbc/blob/master/src/main/java/org/sqlite/core/NativeDB.c#L352

**Comment by tildedave** [comment]
I've included a fix for this in https://github.com/xerial/sqlite-jdbc/pull/456

**Comment by tildedave** [comment]
I believe this will be fixed now that #456 is merged

**Comment by gotson** [comment]
Closing this as it should be fixed. Comment if necessary.
